instrumented/intrumented.go

```diff
@@ -0,0 +1,136 @@
+package instrumented
+
```

Comment:
This file has a typo in the name.  I also wonder whether there is a better name than `instrumented` because it seems quite unclear what it's about when imported (lots of things can be instrumented) and it doesn't reflect that it's about substrate.  I don't have a good suggestion yet though.

Reply:
(I don't feel super strongly about this by the way)

Reply:
What about splitting into two different files `source.go` and `sink.go`?

Reply:
makes sense for the files.  The package name is a different question.